Merge mu branch with main

FlamingGears opened the pull request:
added:
-physics based projectiles using "arc-projectile" template (use same bullet_data format though)
-basic enemies with health that delete themselves when dead
-two types of spawn pads and a basic spawn point
-two new testing levels (separate level scenes that are placed into world)
-a simple sentry gun
-interaction system for buttons (needs refining)

[] note, disabled firing sound in animation and lowered sound volume in file, feel free to revert

FlamingGears committed:
Sentry added

Sentry added with bullets

FlamingGears committed:
Branch Update

added spawner,"fixed" turret, buttons incomplete

FlamingGears committed:
removed .godot from gitignore

For some reason a buch of gtlf or mtl or something files are locaed in the editor in .godot. Hopefully this now imports correctly.

FlamingGears committed:
added systems.

Added enemies, spawn pad/node, simple path finding

FlamingGears committed:
Update script_editor_cache.cfg

FlamingGears committed:
added teleporters, partly fixed ammo,

bullets when created can be told who shot them,
added teleporters that work on channels. only 2 per channel or weirdness happens.

FlamingGears committed:
backup

everything so far, need to get better at using this.

FlamingGears committed:
last push before migration

backing up before I attempt to combine main build into mine.

FlamingGears committed:
last before merge

FlamingGears committed:
Merge branch 'main' into nbreslich-last-before-merge-branch

FlamingGears committed:
merged over branch/disable sound

merged branch, re-created the classic bullet, disabled sound on animation and messed with sound file

FlamingGears committed:
tweaked vent bomb

added safety fuze, added temp ammo pickup
note, need way to carry over player's momenum to bullet fired

FlamingGears committed:
bullets know who shot them

added "orginator" to progectile.gd template, added bullet_node.orginator = self to player to inform bullet,
added "Pbody" to player data so it can share the player's body.

FlamingGears committed:
Added damage to projectiles

uses body.InflictDamage where the object must have the InflictDamage function to take damage,
(we can add  damage reduction this way)

FlamingGears committed:
re-added interaction

same script, modified to fit new node placement. Added label in hud scene for interaction text

FlamingGears committed:
fixed classic bullet sentry

changed the layers, 2 is detection now? had to tweak projectile code slightly, now hits but  not damages player if not enemy bullet. Player now takes damage

FlamingGears committed:
spawnpad tweak + cleanup

spanwpads can now be walked on fully
removed all the mortar bullet code

FlamingGears committed:
added test levels and fixed enemy code

added 2 new test levels so Gabe can modify the original scene without conflicting other scenes.
added load("res://componets/player/player_info.tres").Pbody
to the enemy code so the enemy no longer cares where it is in the node tree

FlamingGears committed:
Finished adding in new levels

Pathfinding level for enemy testing
tool level for testing spawners, teleporters, interactables, etc

FlamingGears committed:
fixed projectile detect glitch

fixed if has(inflict damage) so game should no longer crash when a bullet tries to damage a non-damage able enemy

FlamingGears committed:
heavily modified bullet code

added a physics based bullet
added foundation for explosives

FlamingGears committed:
Last before pull

tweaked values reguarding physics bullet

FlamingGears committed:
New Sentry!

Added a completely new sentry
modified the test levels to well, test it.
added print statements to projectile code
attempt: upgrade bullet detection

FlamingGears committed:
FIXED PROJECTILES

added: or orginator == body.owner):
to 	if (orginator == body or orginator == body.owner):
also added fancy tree to test hitboxes

FlamingGears committed:
added laser bullet

made new beam type projectile
took many an hour

FlamingGears committed:
started creating kinmatics

testing kinmatics

FlamingGears committed:
crawler,,laser, cube

started on crawler, imported a 1x1x3 mesh from onshape to test cad importing,
added emission to laser, intent: glow

Guuzzeji commented:
@FlamingGears, Everything looks good, but could you consider excluding the ".godot" folder content? Currently, you're adding over 3,000 files to the main repository, with only about 100 files being necessary code. I'm concerned this might cause issues on the main branch, leading to unnecessary bloat. I suggest keeping only the in the "editor" folder within the ".godot" folder. If that doesn't work, just push everything into the main branch. We can figure that out later.

Guuzzeji reviewed (approved):
Approve pull request, you can now push changes into main branch

FlamingGears committed:
updated exceptions + proc Amin

updated the exceptions to exclude everything inside the .godot folder except the editor and import folder.

FlamingGears committed:
"finished" crawler

walking animation kinda works.  Should probably only apply code to small enemies.

FlamingGears committed:
added projectile features

hit enemies can detect if weakpoint was hit and if bullet hit multiple hitboxes,
bullets now pass body and self, inflict_damage(damage, body, bulletinstance)

FlamingGears committed:
Removed crawler from spawn + starated tele

removed crawler from the main testing area - no longer needed
- started work on 1-way teleporter for level enter / exits

FlamingGears committed:
ONE-WAY TELEPORTER

AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAa why does godot suffer. Anyway does as expected. 2 nodes that must share channel, one enters one exits

FlamingGears committed:
added light experiment, started prototype

added a scene to test lighting, trying to make a glowing cube. Just disable shading and set color to max saturation. Started on a prototype of what a typical level would look like

FlamingGears committed:
tested_importing/protoLevel

1m cube onshape = 1m cube godot
no scale issues so far!

FlamingGears committed:
Level geometry

imported level geometry and began work on utilities like pressure pads for doors

FlamingGears committed:
more lvl work

FlamingGears committed:
added checks to multiple connect functions

if !Life_Timer.timeout.is_connected(_on_life_timer_timeout):
added this everywhere i could to stop red errors

FlamingGears committed:
deleted unneeded meshes

Two imported cuboids from onshape were moved in files, I deleted them but the files might still remain somewhere.

FlamingGears committed:
delta -> _delta

fixed many unused delta errors by adding _ to the front of them.

FlamingGears committed:
Prototurret now enemy

proto turret now takes damage as it is in group enemy.
enemy is generic group for identification only, specific enemy groups are for specific areas.

FlamingGears committed:
added group enemy check

The level node every frame checks the size of a group, if there are none, emit signal (that is manually connected to a specific door)

FlamingGears committed:
woo colors

added basic colors to protolevel1

FlamingGears committed:
added navmesh to level

moving enemies can now navigate the level, needs work though.

FlamingGears committed:
exit door and teleporter work

Exit door opens and player can leave the level. Just need to reset the level on exit then i can remake the enemies.

FlamingGears committed:
???

not sure what this was,

FlamingGears committed:
ah so it is the chache8

this changes on godot launch, might be useful to save here.

FlamingGears committed:
Began work on dmg numbers

made new scene,
- simple falling + fade text animation,
- started script for it.

FlamingGears committed:
Hud damage numbers

Bullet damage on hit is now displayed on screen. Explanation on google doc

FlamingGears committed:
started new sentry

started work on the Basic Bullet Sentry
- first encounter

FlamingGears committed:
Fixed the projectile bullets

hitting hitboxes now damages what they are attached to!!

FlamingGears committed:
continued work on basic bullet turret

added temporary node tree with placeholder meshes.
It has untested code attached to it.
Code is based on the generic sentry code script.

FlamingGears committed:
disabled print in bullet, added mesh to teleport entery

as title and added a mesh to the teleporter entry for the test level> (kept tripping over it).

FlamingGears committed:
Added raycasts to Basic  Bullet Sentry

FlamingGears committed:
line of sight now working for basic bullet sentry

sentry now only fires if it has a valid target  has entered the activation area and has line of sight

FlamingGears committed:
modeled and animated Basic Bullet Sentry

the sentry now has animations. Had to add collision layer 15 to player model.
Reason: sentry has separate collision mesh to stop player from sticking to it

FlamingGears committed:
added bolt action to BBS

Basic bullet sentry now has a back cap that flies back when firing for added effect.

FlamingGears committed:
Added weakspot functionality to BBS

Should now take double damage from its to weak spot. Currently changes must be hardcoded if this code is reused.
-- also now reports crits and death to player

FlamingGears committed:
Tweaked Damage numbers

Temp change: better plan, 4 separate labels (or even more!) for each type of damage done. Having 1 label creates weird issues with texturing

FlamingGears committed:
made label text local to scene and speed tweak

Fixed a glitch where a dark red crit kill would recolor all other falling damage numbers.
Slowed the number fall somewhat

FlamingGears committed:
added mana to sentry

the mana for the BBS is offset between 0 and -0.19. -0.19

FlamingGears committed:
finishing up BBS

BBS now has working health, mana, damage, animations, etc.
Last tasks:
- new bullet for it
- death animations + state.

FlamingGears committed:
BBS death animation

Added death animation. Player was hard to use but the "collapse in current state" part of the animation was a free bonus from the player.

FlamingGears committed:
redid proto level signal system.

FlamingGears committed:
suppressed print statements

FlamingGears committed:
began work on new sentry

Guardian sentry (really just gatling sentry)
idea is that it shoots in bursts. Reuses much of the basic bullet sentry code

FlamingGears committed:
worked on animations and firing

FlamingGears committed:
added hitboxes to GS

FlamingGears committed:
updated button spawner

can rotate spawned object but does not rotate the correct amount regardless of rad or deg? probably an offset origin issue.

FlamingGears committed:
started work on GS burst fire + OOPS

OOPS accidentally programmed the burst in the basic bullet sentry's script. Should be fixed now but there could be hidden errors.

FlamingGears committed:
guardian sentry now has functional burst

FlamingGears committed:
fixed some errors with "_"

Patched all those "parameter not used" errors.

FlamingGears committed:
tweaked GS sentry

FlamingGears committed:
added barrel spin animation

FlamingGears committed:
sentryProgress

tweaked values for guardian sentry. Still working on death animations

FlamingGears committed:
Finished Guardian sentry

Sentry should be fully functional with only stat changes required. Just place in area and it should work. Might want to change the area detection.

FlamingGears committed:
improved projectiles

ar projectile now functions
ISSUE: bullets now no longer collide as an issue occures where a bullet is que_freed() before the group check sshould go through, causing a crash where a group is requested from a null object

FlamingGears committed:
started work on crawler enemy

FlamingGears committed:
testing crawler code

FlamingGears committed:
crawler now crawls

turns out the little skeleton heads must manually updated when you copy a skeleton to a new scene.

FlamingGears committed:
turning crawler

crawler can now turn on its own using modified turret code. There are anomalies in the walking so i will have to tweak that

FlamingGears committed:
tweaked turning with better code

while the turret code is still used to find turn amount, the leg code is used for the actual turning. Issues could occur as the legs support multi surface climbing though nothing else does.

FlamingGears committed:
tweaking crawler skeleton

FlamingGears committed:
modifying spawn points, button, and crawler

FlamingGears committed:
"fixed" pathfinding of shooty_crawler

The crawler now moves into range and retreats when it is too close. Code is quite janky though and unoptimized.

FlamingGears committed:
added health and damage to crawler

currently does nothing until hitboxes are added

FlamingGears committed:
shooty crawler damage is functioning now

FlamingGears committed:
started combat work for crawler

Enabling crawler to create a sentry hat that shares health with base, functionally creating one enemy.

FlamingGears committed:
making sentry hat for shooty crawler

designing a specialized sentry to be used  as hat by the crawler base. Currently made out of reused basic bullet sentry code and assets

FlamingGears committed:
shooty crawler prototype working

The enemy seeks the player and functions correctly (mostly). Line of sight currently bypassed as it appears broken. Has no animations and needs a new skeleton.

FlamingGears committed:
new "GunGoon" enemy template-ish

Basically the crawler minus any of the crawling. Currently just a template to play around with the pathfinding quirks. Will become a proper enemy later.

FlamingGears committed:
backup 4/1/12024

note: look into the friendly fire issue. At least don't report it to the player?

FlamingGears committed:
added comments and tweaked enemy scripts

mainly tweaked gungoon's detection areas for snappier movement
line of sight is still broken

FlamingGears committed:
modified arc code, warning

The other phys projectiles will need to be updated.
The mortar shell is fine as it was edited for it.

FlamingGears committed:
Working on mortar shell

Added area functions and blast timer.

FlamingGears committed:
Added for loop and if tree to damage function

for mortar shell
